tests/testthat/test-renderer3-HitOrMissMonteCarlo.R

```diff
@@ -0,0 +1,171 @@
+                          title="Hit or Miss Monte Carlo Integration",
+                          time=list(variable="iter", ms=200),
+                          first=list(iter=1)))
+  return(viz.publish)
+}
+
+viz <- HitOrMissMonteCarlo()
+
+info <- animint2HTML(viz)
+
+# Test of needed data columns in TSV files
+chunk1.tsv <- file.path("animint-htmltest", "geom1_path_vizone_chunk1.tsv")
+chunk1 <- read.table(chunk1.tsv, sep="\t", header=TRUE,
+                     comment.char="", quote="")
+
+chunk2.tsv <- file.path("animint-htmltest", "geom2_point_vizone_chunk1.tsv")
+chunk2 <- read.table(chunk2.tsv, sep="\t", header=TRUE,
+                     comment.char="", quote="")
+
+chunk3.tsv <- file.path("animint-htmltest", "geom5_path_viztwo_chunk1.tsv")
+chunk3 <- read.table(chunk3.tsv, sep="\t", header=TRUE,
+                     comment.char="", quote="")
+
+chunk4.tsv <- file.path("animint-htmltest", "geom6_text_viztwo_chunk1.tsv")
+chunk4 <- read.table(chunk4.tsv, sep="\t", header=TRUE,
+                     comment.char="", quote="")
+
+chunk5.tsv <- file.path("animint-htmltest", "geom4_hline_viztwo_chunk1.tsv")
+chunk5 <- read.table(chunk5.tsv, sep="\t", header=TRUE,
+                     comment.char="", quote="")
+
+chunk6.tsv <- file.path("animint-htmltest", "geom3_tallrect_viztwo_chunk1.tsv")
+chunk6 <- read.table(chunk6.tsv, sep="\t", header=TRUE,
+                     comment.char="", quote="")
+
+test_that("Integration Curve data contains expected columns", {
+  path_expected_cols <- c("x", "y", "group")
+  point_expected_cols <- c("shape", "colour", "x", "y", "key", "showSelectedlegendshapecolour", "showSelected1", "fill")
+  
+  expect_identical(sort(names(chunk1)), sort(path_expected_cols))
+  expect_identical(sort(names(chunk2)), sort(point_expected_cols))
+})
+
+test_that("Pi estimation graph data contains expected columns", {
+  path_expected_cols <- c("x", "y", "showSelected1", "key", "group")
+  text_expected_cols <- c("x", "y", "key", "label", "showSelected1", "group")
+  tallrect_expected_cols <- c("xmin", "xmax", "clickSelects", "group")
+  hline_expected_cols <- c("yintercept")
+  
+  expect_identical(sort(names(chunk3)), sort(path_expected_cols))
+  expect_identical(sort(names(chunk4)), sort(text_expected_cols))
+  expect_identical(sort(names(chunk5)), sort(hline_expected_cols))
+  expect_identical(sort(names(chunk6)), sort(tallrect_expected_cols))
+})
+
+test_that("x and y labels are not null", {
+  xlabel <- getNodeSet(info$html, "//text[@class='xtitle']")
+  ylabel <- getNodeSet(info$html, "//text[@class='ytitle']")
+  
+  expect_equal(length(xlabel), 2)
+  expect_equal(length(ylabel), 2)
+})
+
+test_that("Different Points Are Rendered on Integration Curve", {
+  x1_nodes <- getNodeSet(info$html, "//g[@class='geom2_point_vizone']/g[@class='PANEL1']/circle/@cx")
+  y1_nodes <- getNodeSet(info$html, "//g[@class='geom2_point_vizone']/g[@class='PANEL1']/circle/@cy")
+  
+  Sys.sleep(1.739)
```

Comment:
why this number ?

Reply:
I took the reference from other successful tests. `Sys.sleep()` is used to pause the program for the specified time so that next reading can be taken. I used the same time as in other tests.